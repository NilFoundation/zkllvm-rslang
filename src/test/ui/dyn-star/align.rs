// revisions: normal over_aligned
//[normal] check-pass

#![feature(dyn_star)]
//~^ WARN the feature `dyn_star` is incomplete and may not be safe to use and/or cause compiler crashes

use std::fmt::Debug;

#[cfg_attr(over_aligned,      repr(C, align(1024)))]
#[cfg_attr(not(over_aligned), repr(C))]
#[derive(Debug)]
struct AlignedUsize(usize);

fn main() {
    let x = AlignedUsize(12) as dyn* Debug;
    //[over_aligned]~^ ERROR `AlignedUsize` needs to be a pointer-sized type
}
