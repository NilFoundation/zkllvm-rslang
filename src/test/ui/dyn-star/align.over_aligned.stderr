warning: the feature `dyn_star` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/align.rs:4:12
   |
LL | #![feature(dyn_star)]
   |            ^^^^^^^^
   |
   = note: see issue #91611 <https://github.com/rust-lang/rust/issues/91611> for more information
   = note: `#[warn(incomplete_features)]` on by default

error[E0277]: `AlignedUsize` needs to be a pointer-sized type
  --> $DIR/align.rs:15:13
   |
LL |     let x = AlignedUsize(12) as dyn* Debug;
   |             ^^^^^^^^^^^^^^^^ `AlignedUsize` needs to be a pointer-sized type
   |
   = help: the trait `PointerSized` is not implemented for `AlignedUsize`

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0277`.
